pkg/plugin/processor/builtin/convertField.go

```diff
@@ -0,0 +1,169 @@
+// Copyright © 2024 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package builtin
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/conduitio/conduit-commons/opencdc"
+	sdk "github.com/conduitio/conduit-processor-sdk"
+	"github.com/conduitio/conduit/pkg/foundation/cerrors"
+)
+
+type convertField struct {
+	referenceResolver sdk.ReferenceResolver
+	typ               string
+
+	sdk.UnimplementedProcessor
+}
+
+func (p *convertField) Specification() (sdk.Specification, error) {
+	return sdk.Specification{
+		Name:    "field.convert",
+		Summary: "convert the type of a field",
+		Description: "convert takes the field of one type and converts it into another type (e.g. string to integer). " +
+			"The applicable types are string, int, float and bool. Converting can be done between any combination of " +
+			"types. Note that booleans will be converted to numeric values 1 (true) and 0 (false). Processor is only " +
+			"applicable to .Key, .Payload.Before and .Payload.After prefixes, and only applicable if said fields are structured data.",
+		Version: "v1.0",
+		Author:  "Meroxa, Inc.",
+		Parameters: map[string]sdk.Parameter{
+			"field": {
+				Default:     "",
+				Type:        sdk.ParameterTypeString,
+				Description: "the target field, as it would be addressed in a Go template",
+				Validations: []sdk.Validation{
+					{
+						Type: sdk.ValidationTypeRequired,
+					},
+				},
+			},
+			"type": {
+				Default:     "",
+				Type:        sdk.ParameterTypeString,
+				Description: "the target field type after conversion",
+				Validations: []sdk.Validation{
+					{
+						Type: sdk.ValidationTypeRequired,
+					}, {
+						Type:  sdk.ValidationTypeInclusion,
+						Value: "string,int,float,bool",
+					},
+				},
+			},
+		},
+	}, nil
+}
+
+func (p *convertField) Configure(_ context.Context, cfg map[string]string) error {
+	field, ok := cfg["field"]
+	if !ok {
+		return cerrors.Errorf("%w (%q)", ErrRequiredParamMissing, "field")
+	}
+	if !strings.HasPrefix(field, ".Payload.After") &&
+		!strings.HasPrefix(field, ".Payload.Before") &&
```

Comment:
but having the prefix `.Payload` could still result in an error, if the user used some invalid field like `.Payload.Something`, that's why I separated them.

Reply:
In that case, you would still be able to write `!strings.HasPrefix(field, ".Payload")`, because that eliminates everything except the key and payload.

And then below, when you write `resolver, err := sdk.NewReferenceResolver(field)`, you'd get an error (no Something field). It's a nitpick anyway, a few characters saved only.